airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/write/typing_deduping/direct_load_tables/BigqueryDirectLoadSqlGenerator.kt

```diff
@@ -329,10 +330,15 @@ class BigqueryDirectLoadSqlGenerator(
             if (stream.importType is Dedupe) {
                 // We're doing de-duping, therefore we have a primary key.
                 // Cluster on the first 3 PK columns since BigQuery only allows up to 4 clustering
-                // columns,
-                // and we're always clustering on _airbyte_extracted_at
+                // columns, and we're always clustering on _airbyte_extracted_at
                 (stream.importType as Dedupe).primaryKey.stream().limit(3).forEach {
                     pk: List<String> ->
+                    val bigqueryType = toDialectType(stream.schema.asColumns()[pk.first()]!!.type)
```

Comment:
You might want to break this up onto multiple lines so the various NPEs are easier to differentiate